plugins/autolink/plugin.js

```diff
@@ -73,38 +73,29 @@
 				}
 			}
 
-			function tryToEncodeLink( data ) {
-				// If enabled use link plugin to encode email link.
-				if ( editor.plugins.link ) {
-					var link = CKEDITOR.dom.element.createFromHtml( data ),
-						linkData = CKEDITOR.plugins.link.parseLinkAttributes( editor, link ),
-						attributes = CKEDITOR.plugins.link.getLinkAttributes( editor, linkData );
+			function getHtmlToInsert( text ) {
+				var link = new CKEDITOR.dom.element( 'a' ),
+					value = text.replace( /"/g, '%22' );
 
-					if ( !CKEDITOR.tools.isEmpty( attributes.set ) ) {
-						link.setAttributes( attributes.set );
-					}
+				value = value.match( CKEDITOR.config.autolink_urlRegex ) ? value : 'mailto:' + value;
 
-					if ( attributes.removed.length ) {
-						link.removeAttributes( attributes.removed );
-					}
+				link.setText( text );
+				link.setAttribute( 'href', value );
 
-					link.removeAttribute( 'data-cke-saved-href' );
+				var linkData = CKEDITOR.plugins.link.parseLinkAttributes( editor, link ),
```

Comment:
It isn't absolutely necessary, but let's add a ticket reference above this line as it isn't present anywhere yet.

Reply:
I added issue reference.